packages/react/src/context/slash-id-context.tsx

```diff
@@ -201,25 +205,42 @@ export const SlashIDProvider = ({
                 value: handle.value,
               };
 
-        const user = await sid
-          // @ts-expect-error TODO make the identifier optional
-          .id(oid, identifier, factor)
-          .then(async (user) => {
-            return applyMiddleware({ user, sid, middleware });
-          });
+        const newlyLoggedInUser = await (async () => {
+          if (user && userIsAnonymous(user)) {
+            // @ts-expect-error TODO make the identifier optional
+            return await user.id(oid, identifier, factor).then(async (user) => {
+              return applyMiddleware({ user, sid, middleware });
+            });
+          }
+
+          return await sid
+            // @ts-expect-error TODO make the identifier optional
+            .id(oid, identifier, factor)
+            .then(async (user) => {
+              return applyMiddleware({ user, sid, middleware });
+            });
+        })();
+
+        storeUser(newlyLoggedInUser);
 
-        storeUser(user);
         return user;
       } catch (e) {
         logOut();
         throw e;
       }
     },
-    [oid, state, storeUser, logOut]
+    [state, storeUser, user, oid, logOut]
   );
 
   const mfa = useCallback<MFA>(
     async ({ handle, factor }) => {
+      if (user && userIsAnonymous(user)) {
```

Comment:
We need the same check on the `recover` method as it cannot be called with an anonymous user.